ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1108,6 +1109,17 @@ public CompletableFuture<RaftClientReply> setConfigurationAsync(SetConfiguration
       if (arguments.getMode() == SetConfigurationRequest.Mode.ADD) {
         serversInNewConf = add(RaftPeerRole.FOLLOWER, current, arguments);
         listenersInNewConf = add(RaftPeerRole.LISTENER, current, arguments);
+      } else if (arguments.getMode() == SetConfigurationRequest.Mode.CAS) {
+        List<RaftPeer> serversInCurConf = arguments.getServersInCurConf();
+        List<RaftPeer> listenersInCurConf = arguments.getListenersInCurConf();
+        if (serversInCurConf.equals(current.getAllPeers(RaftPeerRole.FOLLOWER))
```

Comment:
When two lists are in different orders, they become unequal.  We need an equalsIgnoreOrder method.

Reply:
Good catch! I have added the method and changed the related code.